text/0000-input-macros.md

```diff
@@ -0,0 +1,484 @@
+- Feature Name: input_macros
+- Start Date: 2025-04-19
+- RFC PR: [rust-lang/rfcs#3799](https://github.com/rust-lang/rfcs/pull/3799)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This RFC proposes the addition of macros or some functions that can be used to 
+read input from the user in a more ergonomic way, similar to the 
+[Input built-in function in Python](https://peps.python.org/pep-3111/). 
+
+With this initiative, we can build a small interactive programs that reads input 
+from standard input and writes output to standard output. This is 
+well-established as a simple and fun way of learning and teaching Rust as a 
+new programming language. 
+
+```rust
+println!("Please enter your name: ");
+let possible_name: Result<String, _> = input!(); // This could fail, for example 
+                                                 // if the user closes the input 
+                                                 // stream
+
+// Besides we can show a message to the user
+let possible_age: Result<u8, _> = input!("Please enter your age: "); 
+                                        // This could fail, for example if the 
+                                        // user enters a string instead of a 
+                                        // number in the range of u8
+
+// And yes, this is a result so we can handle errors like this
+let lastname: String = input!("Please enter your lastname: ")
+                        .expect("The lastname is required"); 
+                            // This could fail for example if the 
+                            // user enters a empty string
+
+// --- Another way to use the macro ---
+
+struct Price {
+    currency: String,
+    amount: f64,
+}
+
+impl FromStr for Price {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let parts: Vec<&str> = s.split_whitespace().collect();
+        if parts.len() != 2 {
+            return Err("String must have two parts".to_string());
+        }
+        let currency = parts[0].to_string();
+        let amount = parts[1].parse().unwrap();
+        Ok(Price { currency, amount })
+    }
+}
+
+let price: Price = input!("Please enter a price: ")?; 
+        // This could fail for example if the input is reading from a pipe and 
+        // we delete the file whose descriptor is being read while the
+        // program is running
+
+```
+
+The examples above demonstrate several ways to use the `input!` macro.
+
+In this macro, reaching EOF is consider an error case, so we return a `Result` 
+with an error type indicating the cause of the EOF. This approach makes error 
+handling straightforward and maintains consistent behavior.
+
+Alternatively, the following can be used:
+
+```rust
+let name: Option<Price> = try_input!("Please enter a price: ")?;
+```
```

Comment:
I don't think this adds enough value to keep in this proposal.  Handling EOF specially can already be done by checking for the proposed `InputError::Eof`, or by using the underlying primitives.

Reply:
Yes, I had the same impression. I added it just for the sake of correctness and because there were some comments about it.
If you agree, this idea could be discarded in my opinion.
I don't think it's a bad idea in those terms, but at the same time, I feel it wouldn't add enough value and could complicate the discussion.